chutney/server/src/main/java/com/chutneytesting/security/infra/jwt/JwtUtil.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * SPDX-FileCopyrightText: 2017-2024 Enedis
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package com.chutneytesting.security.infra.jwt;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.io.Decoders;
+import io.jsonwebtoken.security.Keys;
+import java.util.Date;
+import java.util.Map;
+import java.util.function.Function;
+import javax.crypto.SecretKey;
+import org.springframework.security.core.userdetails.UserDetails;
+import io.jsonwebtoken.Jwts;
+
+
+public class JwtUtil {
+
+    private final String SECRET_KEY;
+    private final int TOKEN_VALIDITY;
+
+    public JwtUtil(JwtUtilPropertyConfiguration jwtUtilPropertyConfiguration) {
+        this.SECRET_KEY = jwtUtilPropertyConfiguration.getSecretKey();
+        this.TOKEN_VALIDITY = jwtUtilPropertyConfiguration.getTokenValidityInHours();
+    }
+
+    public String generateToken(String username, Map<String, Object> claims) {
+        return createToken(claims, username);
+    }
+
+    public String extractUsername(String token) {
+        return extractClaim(token, Claims::getSubject);
+    }
+
+    public <T> T extractClaim(String token, Function<Claims, T> claimsResolver) {
+        final Claims claims = extractAllClaims(token);
+        return claimsResolver.apply(claims);
+    }
+
+    public boolean validateToken(String token, UserDetails userDetails) {
+        final String username = extractUsername(token);
+        return (username.equals(userDetails.getUsername()) && !isTokenExpired(token));
+    }
+
+
+    private boolean isTokenExpired(String token) {
+        return extractExpiration(token).before(new Date());
+    }
+
+    private Date extractExpiration(String token) {
+        return extractClaim(token, Claims::getExpiration);
+    }
+
+    private String createToken(Map<String, Object> claims, String subject) {
```

Comment:
Why not forge a token with an local issuer for chutney ? and add chutney authorizations as claims/scopes ?